nixos/modules/services/web-servers/fcgiwrap.nix

```diff
@@ -20,28 +20,44 @@ in {
         description = "Number of processes to prefork.";
       };
 
-      socketType = mkOption {
-        type = types.enum [ "unix" "tcp" "tcp6" ];
-        default = "unix";
-        description = "Socket type: 'unix', 'tcp' or 'tcp6'.";
-      };
-
       socketAddress = mkOption {
         type = types.str;
         default = "/run/fcgiwrap.sock";
         example = "1.2.3.4:5678";
-        description = "Socket address. In case of a UNIX socket, this should be its filesystem path.";
+        description = ''
+          Socket address as defined in
+          <citerefentry><refentrytitle>systemd.socket</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+          for <literal>ListemStream</literal>.
+        '';
+      };
+
+      socketUser = mkOption {
+        type = types.str;
+        default = "root";
+        description = "Owner of the socket if it is defined as a Unix socket.";
+      };
+
+      socketGroup = mkOption {
+        type = types.str;
+        default = "root";
+        description = "Group of the socket if it is defined as a Unix socket.";
+      };
+
+      socketMode = mkOption {
+        type = types.str;
+        default = "0660";
+        description = "File mode of the socket if it is defined as a Unix socket.";
```

Comment:
I suggest we don't bikeshed this PR on this discussion, but…

Would it maybe make sense to make a `socket` type which contains all these options, since it may well be used in multiple places? It doesn't seem that nice to have these settings which are essentially aliases for `systemd.sockets.fcgiwrap.socketConfig.*`.